src/timeout.ts

```diff
@@ -39,6 +39,7 @@ export class Timeout extends Promise<never> {
   public ended: number | null = null;
   public duration: number;
   public timedOut = false;
+  public cleared = false;
```

Comment:
Add cleared flag that gets set on `Timeout.clear` so that the TimeoutContext can know if it needs to reconstruct a Timeout for retrying operations